src/symbol/shaping.js

```diff
@@ -39,22 +42,109 @@ export type Shaping = {
 type SymbolAnchor = 'center' | 'left' | 'right' | 'top' | 'bottom' | 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right';
 type TextJustify = 'left' | 'center' | 'right';
 
-function breakLines(text: string, lineBreakPoints: Array<number>) {
+class TaggedString {
+    text: string;
+    sectionIndex: Array<number>
```

Comment:
`// maps each character in 'text' to its corresponding entry in 'sections'`

Reply:
🤔 The `Formatted` and `TaggedString` types kind of evolved separately as I worked on the implementation, but in the final result they end up looking pretty darn similar. I wonder if it makes sense to combine them? I don't think the cost of converting between the two is a big deal, and `TaggedString` has a lot of shaping-focused details, so I guess my inclination is to leave as is (but also probably I'm just lazy so good to get another pair of eyes on the decision).